### PR TITLE
Enhancement: Type fixes

### PR DESCRIPTION
**Pull request by Amadeus-Winarto**
### Summary 
- Refactored contentful related API calls for easier maintenance
- Improved type checks for Events and Admissions

### Details
- All functions which make API calls to Contentful are under `utils/contentful`. The functions are categorised based on the objects in which they operate on e.g. Event, Admission, etc. 
- Objects retrieved from Contentful are always of type Event<T>, where T is a parameter type related to the content type defined in Contentful. These are defined under `utils/contentful/types.ts` for better type checking

In the future, when making a new function to call Contentful's APIs, they are to be added to the folder `utils/contentful`, and the appropriate types are to be defined in `utils/contentful/types.ts`.
 
### Checks 
- [x] Simon
- [ ] Bernard

### CODE REVIEW

File: src/components/Page/index.tsx [hunk 1]
```diff
@@ -105,6 +85,7 @@ const Page: React.FC<OwnProps> = ({
   const urlMap = new Map(
     data.map((image) => [image.fields.title, image.fields.file.url])
   );
+  console.log(data);
```
Comment (simonjulianl) [inline src/components/Page/index.tsx:88]: Remove this naughty line

**Review by simonjulianl — CHANGES_REQUESTED** [review]
ada console.log

**Comment by Amadeus-Winarto** [comment]
> ada console.log

FUDGE benar jg sabar

File: src/components/Page/index.tsx (hunk 1 above)
Comment (Amadeus-Winarto) [reply src/components/Page/index.tsx:88]: yes boss

**Review by simonjulianl — APPROVED** [review]
LGTM

**Pull request merged**